llarp/messages/relay_status.hpp

```diff
@@ -0,0 +1,107 @@
+#ifndef LLARP_RELAY_STATUS_HPP
+#define LLARP_RELAY_STATUS_HPP
+
+#include <crypto/encrypted_frame.hpp>
+#include <crypto/types.hpp>
+#include <messages/link_message.hpp>
+#include <path/path_types.hpp>
+#include <pow.hpp>
+
+#include <array>
+#include <memory>
+
+namespace llarp
+{
+  // forward declare
+  struct AbstractRouter;
+  namespace path
+  {
+    struct PathContext;
+    struct IHopHandler;
+  }  // namespace path
+
+  struct LR_StatusRecord
+  {
+    static constexpr uint64_t SUCCESS               = 1;
```

Comment:
I would make this an enum class instead, extending `uint64_t` ( and possibly overloading `operator|(Status status)`).

The status is a closed sets of discrete possibilities. It makes no sense to go outside the range of the specified enumerator, or to combine them with integers. Therefore, they are a "true enum".

In general, you wouldn't compare a `Status` to an integer - it makes sense that this operation is disallowed by default. By having to use a static_cast, we make it clear that we are breaking the abstraction of "closed set of possibilities". I like that clarity, even if results in slightly more code.

Furthermore, an explicit cast reduces the chance of bugs caused by implicit conversions.